models/DataObject/ClassDefinition/Data/FieldDefinitionEnrichmentInterface.php

```diff
@@ -20,6 +20,11 @@
 interface FieldDefinitionEnrichmentInterface
 {
     /**
+     * If in admin mode this method can be implemented to change the fielddefinition whenever
+     * getFieldDefinition() get called on the data type.
+     * One example purpose is to populate or change dynamic settings like the options for select and multiselect fields.
+     * The context param contains contextual information about the container, the field name, etc ...
```

Comment:
Can we define some contextual info used by internal calls?

Reply:
What would be the use case ?